src/browser/modules/Main/Main.jsx

```diff
@@ -72,11 +75,20 @@ const Main = React.memo(function Main (props) {
           &nbsp; to establish connection. There's a graph waiting for you.
         </NotAuthedBanner>
       </Render>
-      <Render if={props.connectionState === PENDING_STATE}>
+      <Render
+        if={
+          props.connectionState === PENDING_STATE && !showSlowConnectionBanner
+        }
+      >
         <WarningBanner data-testid='reconnectBanner'>
           Connection to server lost. Reconnecting...
         </WarningBanner>
       </Render>
+      <Render if={showSlowConnectionBanner}>
+        <WarningBanner data-testid='reconnectBanner'>
+          Server is taking long to respond...
```

Comment:
I am not sure this message makes sense. 'Server is taking a long time to respond' might be better?

Reply:
Agreed, updated!